pkg/agent/flowexporter/exporter/exporter_test.go

```diff
@@ -247,43 +245,190 @@ func getElemList(ianaIE []string, antreaIE []string) []*ipfixentities.InfoElemen
+		tuple, revTuple = makeTuple(&srcIP, &dstIP, 6, 65280, 255)
+	}
+	conn := &flowexporter.Connection{
+		StartTime:                     time.Time{},
+		StopTime:                      time.Time{},
+		OriginalPackets:               0xab,
+		OriginalBytes:                 0xabcd,
+		ReversePackets:                0xa,
+		ReverseBytes:                  0xab,
+		TupleOrig:                     tuple,
+		TupleReply:                    revTuple,
+		IsPresent:                     isPresent,
+		SourcePodNamespace:            "ns",
+		SourcePodName:                 "pod",
+		DestinationPodNamespace:       "",
+		DestinationPodName:            "",
+		IngressNetworkPolicyName:      "",
+		IngressNetworkPolicyNamespace: "",
+		EgressNetworkPolicyName:       "np",
+		EgressNetworkPolicyNamespace:  "np-ns",
+		DestinationServicePortName:    "service",
+	}
+	return conn
+}
+
+func getFlowRecord(conn *flowexporter.Connection, isIPv6 bool, isActive bool) flowexporter.FlowRecord {
+	flowRecord := &flowexporter.FlowRecord{
+		Conn:               *conn,
 		PrevPackets:        0,
 		PrevBytes:          0,
 		PrevReversePackets: 0,
 		PrevReverseBytes:   0,
 		IsIPv6:             isIPv6,
+		IsActive:           isActive,
+	}
+	return *flowRecord
+}
+
+func TestFlowExporter_sendFlowRecords(t *testing.T) {
+	for _, tc := range []struct {
+		v4Enabled bool
+		v6Enabled bool
+	}{
+		{true, false},
+		{false, true},
+		{true, true},
+	} {
+		testSendFlowRecords(t, tc.v4Enabled, tc.v6Enabled)
+	}
+}
+
+func testSendFlowRecords(t *testing.T, v4Enabled bool, v6Enabled bool) {
+	var elemListv4, elemListv6 []*ipfixentities.InfoElementWithValue
+	if v4Enabled {
+		elemListv4 = getElemList(IANAInfoElementsIPv4, AntreaInfoElementsIPv4)
+	}
+	if v6Enabled {
+		elemListv6 = getElemList(IANAInfoElementsIPv6, AntreaInfoElementsIPv6)
+	}
+	flowExp := &flowExporter{
+		elementsListv4:    elemListv4,
+		elementsListv6:    elemListv6,
+		templateIDv4:      testTemplateIDv4,
+		templateIDv6:      testTemplateIDv6,
+		v4Enabled:         true,
+		activeFlowTimeout: testActiveFlowTimeout,
+		idleFlowTimeout:   testIdleFlowTimeout,
+	}
+
+	if v4Enabled {
+		runSendFlowRecordTests(t, flowExp, false)
+	}
+	if v6Enabled {
+		runSendFlowRecordTests(t, flowExp, true)
+	}
+}
+
+func runSendFlowRecordTests(t *testing.T, flowExp *flowExporter, isIPv6 bool) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockIPFIXExpProc := ipfixtest.NewMockIPFIXExportingProcess(ctrl)
+	mockDataSet := ipfixtest.NewMockIPFIXSet(ctrl)
+	mockConnStore := connectionstest.NewMockConnectionStore(ctrl)
+	flowExp.process = mockIPFIXExpProc
+	flowExp.ipfixSet = mockDataSet
+	flowExp.connStore = mockConnStore
+
+	tests := []struct {
+		name               string
+		isConnPresent      bool
+		isRecordActive     bool
+		packetDifference   uint64
+		lastExportTimeDiff time.Duration
+	}{
+		{
+			"active flow record",
+			true,
+			true,
+			0x2, // non-zero number for active records
```

Comment:
why hexadecimal number for this? Isn't it a packet count?

Reply:
Yes, that is just a packet count. No special reason. I was trying with a higher number before. This can be changed for readability. Will remember in a follow-up PR. Merging this PR now.